tests/general/views/test_consolidated_profile_view.py

```diff
@@ -0,0 +1,566 @@
+        ]
+
+        indicator = self.create_versioned_data(
+            version1, geography, profile, data=data
+        )
+        self.create_highlight(
+            profile, indicator, "female", "highlight_v1"
+        )
+
+        # Version 2 data
+        version2, geography, hierarchy, profile = self.create_base_data(
+            "v2", geography, hierarchy, profile
+        )
+        data = [
+            {"gender": "male", "age": "15", "count": 2},
+            {"gender": "female", "age": "16", "count": 3},
+        ]
+        indicator = self.create_versioned_data(
+            version2, geography, profile, data=data
+        )
+        self.create_highlight(
+            profile, indicator, "female", "highlight_v2"
+        )
+
+        # Request without any version
+        response = self.get(
+            'all-details',
+            profile_id=profile.pk,
+            geography_code=geography.code,
+            data={
+                'format': 'json',
+            },
+        )
+        self.assert_http_200_ok()
+        assert response.data["profile"]["highlights"] == [
+            {'label': 'highlight_v1', 'value': 2.0, 'method': 'absolute_value'}
+        ]
+
+        # Request with version2
+        response = self.get(
+            'all-details',
+            profile_id=profile.pk,
+            geography_code=geography.code,
+            data={
+                'version': version2.name,
+                'format': 'json',
+            },
+        )
+        self.assert_http_200_ok()
+        assert response.data["profile"]["highlights"] == [
+            {'label': 'highlight_v2', 'value': 3.0, 'method': 'absolute_value'}
+        ]
+
+    def test_highlight_sibling_with_multiversion(self):
+        version2 = VersionFactory()
+        geo3 = self.root.add_child(code="GT")
+        self.create_boundary(geo3, version2)
+        self.create_boundary(self.geo1, version2)
+        configuration = self.hierarchy.configuration
+        configuration["versions"] = [self.version.name, version2.name]
+        self.hierarchy.configuration = configuration
+        self.hierarchy.save()
+
+        # self.version data
+        data = {
+            "geo1": [{"gender": "female", "age": "16", "count": 2}],
+            "geo2": [{"gender": "female", "age": "16", "count": 4}],
+            "geo3": [{"gender": "female", "age": "16", "count": 6}],
+        }
+
+        dataset1 = self.create_dataset(self.profile, self.version)
+        dataset2 = self.create_dataset(self.profile, version2)
+        for dataset in [dataset1, dataset2]:
+            self.multiple_datasetdata(dataset, self.geo1, data["geo1"])
+            self.multiple_datasetdata(dataset, self.geo2, data["geo2"])
+            self.multiple_datasetdata(dataset, geo3, data["geo3"])
+            indicator = self.create_indicator(dataset, "gender")
+            self.create_highlight(
+                self.profile, indicator, "female", "highlight_v1", "sibling"
```

Comment:
highlight_v1 doesn't make sense when we create it for both a v1 and a v2 dataset, right?

Reply:
Actually, it makes sense to have the same label of highlight for a different version.
```
dataset of v1 -> indicator of v1 -> highlight
 dataset of v2 -> indicator of v2 -> highlight
```
Both of the highlights should have the same label, So in the frontend, only values of highlight change according to version but the label remains same. 
This was my knowledge of how we handled highlight for diff versions at that point. Not sure if it has changed or I misunderstood how it worked on frontend
 
 But will change it in tests for more readability.

Reply:
Done

Reply:
having the same name was fine, the issue was that the name had the suffix _v1 for both the v1 and the v2 dataset.